SourceX/DiabloUI/text_draw.cpp

```diff
@@ -41,15 +41,14 @@ void DrawTTF(const char *text, const SDL_Rect &rectIn, int flags,
 	if (font == NULL || text == NULL || *text == '\0')
 		return;
 	if (*render_cache == NULL) {
-		*render_cache = new TtfSurfaceCache();
 		const auto x_align = XAlignmentFromFlags(flags);
-		(*render_cache)->text = RenderUTF8_Solid_Wrapped(font, text, text_color, rect.w, x_align);
-		ScaleSurfaceToOutput(&(*render_cache)->text);
-		(*render_cache)->shadow = RenderUTF8_Solid_Wrapped(font, text, shadow_color, rect.w, x_align);
-		ScaleSurfaceToOutput(&(*render_cache)->shadow);
+		*render_cache = new TtfSurfaceCache {
+			/*.text=*/ScaleSurfaceToOutput(SDLSurfaceUniquePtr { RenderUTF8_Solid_Wrapped(font, text, text_color, rect.w, x_align) }),
+			/*.shadow=*/ScaleSurfaceToOutput(SDLSurfaceUniquePtr { RenderUTF8_Solid_Wrapped(font, text, shadow_color, rect.w, x_align) }),

```

Comment:
Why do we have the exact same line twice? If it is needed, can we avoid the code duplication somehow?

Reply:
The lines are different - they use different colors
Refactoring them is out-of-scope for this PR